### PR TITLE
Clean tests to fit new Edit Command, and AB3 remnants

### PR DESCRIPTION
**Pull request by qreoct**
- Another pass to change remaining `Person`s to `StudySpot`s, etc.
- Replaced default test data (like Amy, Bob) to more appropriate Study Spots instead.

Part of #40

### CODE REVIEW

File: src/test/data/JsonSerializableStudyTrackerTest/typicalStudySpotsStudyTracker.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "_comment": "AddressBook save file which contains the same Person values as in TypicalPersons#getTypicalAddressBook()",
+  "_comment": "AddressBook save file which contains the same StudySpot values as in TypicalStudySpots#getTypicalAddressBook()",
```
Comment (limdanqi) [inline src/test/data/JsonSerializableStudyTrackerTest/typicalStudySpotsStudyTracker.json:2]: I think the AddressBook here can change to StudyTracker

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 2]
```diff
@@ -77,59 +77,59 @@ public void parse_invalidPreamble_failure() {
...
         //  No invalid emails
         //  assertParseFailure(parser, " spot/Test" + INVALID_EMAIL_DESC, Email.MESSAGE_CONSTRAINTS); // invalid email
         assertParseFailure(parser, " spot/Test" + INVALID_ADDRESS_DESC,
                 Address.MESSAGE_CONSTRAINTS); // invalid address
         assertParseFailure(parser, " spot/Test" + INVALID_TAG_DESC, Tag.MESSAGE_CONSTRAINTS); // invalid tag
 
-        // invalid phone followed by valid email
-        assertParseFailure(parser, " spot/Test" + INVALID_RATING_DESC + EMAIL_DESC_AMY,
+        // invalid rating followed by valid email
+        assertParseFailure(parser, " spot/Test" + INVALID_RATING_DESC + EMAIL_DESC_FRONTIER,
```
Comment (joenzkimchan) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:88]: One thing we could perhaps do is abstract out the " spot/Test" into a VALID_EDIT_DESC

File: src/test/java/seedu/address/model/ModelManagerTest.java [hunk 3]
```diff
@@ -78,12 +78,12 @@ public void hasStudySpot_nullStudySpot_throwsNullPointerException() {
     }
 
     @Test
-    public void hasStudySpot_personNotInAddressBook_returnsFalse() {
+    public void hasStudySpot_studySpotNotInAddressBook_returnsFalse() {
```
Comment (limdanqi) [inline src/test/java/seedu/address/model/ModelManagerTest.java:81]: AddressBook here should be StudyTracker, same for line 86

**Review by joenzkimchan — APPROVED** [review]
LGTM! Just one small thing we should work towards which I have left as a comment. Thanks for cleaning up the code!

File: src/test/java/seedu/address/model/StudyTrackerTest.java [hunk 4]
```diff
@@ -38,18 +38,18 @@ public void resetData_null_throwsNullPointerException() {
 
     @Test
     public void resetData_withValidReadOnlyAddressBook_replacesData() {
```
Comment (limdanqi) [inline src/test/java/seedu/address/model/StudyTrackerTest.java:40]: There are a few instances of AddressBook found in this class, should be changed to StudyTracker

File: src/test/java/seedu/address/storage/JsonStudyTrackerStorageTest.java [hunk 5]
```diff
@@ -63,7 +63,7 @@ public void readAddressBook_invalidAndValidStudySpotAddressBook_throwDataConvers
     @Test
     public void readAndSaveAddressBook_allInOrder_success() throws Exception {
```
Comment (limdanqi) [inline src/test/java/seedu/address/storage/JsonStudyTrackerStorageTest.java:64]: Instances of AddressBook found in this class, can be changed

File: src/test/java/seedu/address/logic/commands/ClearCommandTest.java [hunk 6]
```diff
@@ -22,8 +22,8 @@ public void execute_emptyAddressBook_success() {
 
     @Test
     public void execute_nonEmptyAddressBook_success() {
-        Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
-        Model expectedModel = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+        Model model = new ModelManager(getTypicalStudyTracker(), new UserPrefs());
```
Comment (limdanqi) [inline src/test/java/seedu/address/logic/commands/ClearCommandTest.java:25]: AddressBook in both test methods can be changed

File: src/test/java/seedu/address/storage/StorageManagerTest.java [hunk 7]
```diff
@@ -54,7 +54,7 @@ public void addressBookReadSave() throws Exception {
          * {@link JsonStudyTrackerStorage} class.
          * More extensive testing of UserPref saving/reading is done in {@link JsonStudyTrackerStorageTest} class.
          */
-        StudyTracker original = getTypicalAddressBook();
+        StudyTracker original = getTypicalStudyTracker();
```
Comment (limdanqi) [inline src/test/java/seedu/address/storage/StorageManagerTest.java:57]: Instances of AddressBook found in this class, can be changed

File: src/test/java/seedu/address/testutil/AddressBookBuilder.java [hunk 8]
```diff
@@ -23,8 +23,8 @@ public AddressBookBuilder(StudyTracker studyTracker) {
     /**
      * Adds a new {@code StudySpot} to the {@code StudyTracker} that we are building.
      */
-    public AddressBookBuilder withStudySpot(StudySpot person) {
-        studyTracker.addStudySpot(person);
+    public AddressBookBuilder withStudySpot(StudySpot studySpot) {
```
Comment (limdanqi) [inline src/test/java/seedu/address/testutil/AddressBookBuilder.java:26]: Can be changed to StudyTracker

**Review by limdanqi — APPROVED** [review]
Looks great except for some minor issues.

**Pull request merged**